conans/client/generators/compiler_args.py

```diff
@@ -0,0 +1,79 @@
+from conans.model import Generator
+from conans.paths import BUILD_INFO_COMPILER_ARGS
+from conans.client.build.compiler_flags import (architecture_flag, sysroot_flag,
+                                                format_defines, format_include_paths,
+                                                format_library_paths, format_libraries,
+                                                build_type_flag, build_type_define, libcxx_flag,
+                                                libcxx_define, rpath_flags, visual_runtime)
+
+
+class CompilerArgsGenerator(Generator):
+
+    @property
+    def filename(self):
+        return BUILD_INFO_COMPILER_ARGS
+
+    @property
+    def compiler(self):
+        return self.conanfile.settings.get_safe("compiler")
+
+    @property
+    def content(self):
+        """With compiler_args you can invoke your compiler:
+        $ gcc main.c @conanbuildinfo.args -o main
+        $ clang main.c @conanbuildinfo.args -o main
+        $ cl /EHsc main.c @conanbuildinfo.args
+        """
+        flags = []
+        flags.extend(format_defines(self._deps_build_info.defines, compiler=self.compiler))
+        flags.extend(format_include_paths(self._deps_build_info.include_paths, compiler=self.compiler))
+
+        flags.extend(self._deps_build_info.cppflags)
+        flags.extend(self._deps_build_info.cflags)
+
+        arch_flag = architecture_flag(arch=self.conanfile.settings.get_safe("arch"), compiler=self.compiler)
+        if arch_flag:
+            flags.append(arch_flag)
+
+        build_type = self.conanfile.settings.get_safe("build_type")
+        btf = build_type_flag(compiler=self.compiler, build_type=build_type)
+        if btf:
+            flags.append(btf)
+        btd = build_type_define(build_type=build_type)
+        if btd:
+            flags.append(format_defines([btd], self.compiler)[0])
```

Comment:
may be better to use extend here for the generic case? like if in future build_type_define will produce multiple definitions (e.g. DEBUG + _DEBUG)?